### PR TITLE
o/configstate/configcore: enable and start prompting handlers

### PR DESCRIPTION
**Pull request by bboozzoo**
Attempt to enable and start prompting handlers whenever prompting is enabled.

Based on top of #14342 

Related: SNAPDENG-22972

### CODE REVIEW

File: overlord/configstate/configcore/prompting.go [hunk 1]
```diff
@@ -36,6 +39,51 @@ import (
...
+	st.Lock()
+	// TODO hookstate? hwo to resolve conflicts
+	tts, err := servicestateControl(st, handlers, inst, nil, &servicestate.Flags{}, nil)
+	st.Unlock()
+	if err != nil {
+		return err
+	}
+
+	st.Lock()
+	chg := st.NewChange("service-control", fmt.Sprintf("Enable and start prompt handlers from snaps: %v", affectedSnaps))
```
Comment (pedronis) [inline overlord/configstate/configcore/prompting.go:70]: s/handlers/handler services/

**Review by pedronis — APPROVED** [review]
thanks, small comment

File: overlord/configstate/configcore/prompting.go [hunk 2]
```diff
@@ -36,6 +39,51 @@ import (
...
+		Action: "start",
+		StartOptions: client.StartOptions{
+			Enable: true,
+		},
+
+		Scope: []string{"user"},
+	}
+
+	st.Lock()
+	// TODO hookstate? hwo to resolve conflicts
```
Comment (olivercalder) [inline overlord/configstate/configcore/prompting.go:62]: s/hwo/how/

File: overlord/configstate/configcore/prompting.go [hunk 3]
```diff
@@ -36,6 +39,51 @@ import (
...
+	tts, err := servicestateControl(st, handlers, inst, nil, &servicestate.Flags{}, nil)
```
Comment (olivercalder) [inline overlord/configstate/configcore/prompting.go:63]: Not sure what `tts` stands for here

File: overlord/configstate/configcore/prompting.go [hunk 4]
```diff
@@ -36,6 +39,51 @@ import (
...
+	st.Unlock()
+	if err != nil {
+		return err
+	}
+
+	st.Lock()
+	chg := st.NewChange("service-control", fmt.Sprintf("Enable and start prompt handlers from snaps: %v", affectedSnaps))
+	for _, ts := range tts {
+		chg.AddAll(ts)
+	}
```
Comment (olivercalder) [inline overlord/configstate/configcore/prompting.go:73]: Same here with `ts`

**Review by olivercalder — APPROVED** [review]
Looks good, thanks! Nitpick about variable names, might be nice if they were a bit more descriptive so it's clear the mechanisms snapd uses around starting handler services

File: overlord/configstate/configcore/prompting.go (hunk 3 above)
Comment (bboozzoo) [reply overlord/configstate/configcore/prompting.go:63]: task sets :) idk why but it's fairly commonly used in the code this way

Comment (olivercalder) [reply overlord/configstate/configcore/prompting.go:63]: I thought it might be task sets, but thought that would be `tss` :shrug: this is fine

**Pull request merged**